pkg/Microsoft.NETCore.Platforms/readme.md

```diff
@@ -0,0 +1,110 @@
+#Runtime IDs
+The package `Microsoft.NETCore.Platforms` defines the runtime identifiers (RIDs) used by .NET packages to represent runtime-specific assets in NuGet packages.
+
+##What is a RID?
+A RID is an opaque string that identifies a platform.  RIDs have relationships to other RIDs by "importing" the other RID.  In that way a RID is a directed graph of compatible RIDs.
+
+##How does NuGet use RIDs?
+When NuGet is deciding which assets to use from a package and which packages to include NuGet will consider a RID if the project.json lists a RID in its `runtimes` section.
+
+- NuGet chooses the best RID-specific asset, where best is determined by a breadth first traversal of the RID graph.  Breadth ordering is document order.
+- NuGet considers RID-specific assets for two asset types: lib and native.
+- NuGet never considers RID-specific assets for compile.
+
+###Best RID
+Consider the partial RID-graph:
+```
+        "any": {},
+
+        "win": {
+            "#import": [ "any" ]
+        },
+        "win-x86": {
+            "#import": [ "win" ]
+        },
+        "win-x64": {
+            "#import": [ "win" ]
+        },
+        "win7": {
+            "#import": [ "win" ]
+        },
+        "win7-x86": {
+            "#import": [ "win7", "win-x86" ]
+        },
+        "win7-x64": {
+            "#import": [ "win7", "win-x64" ]
+        }
+```
+
+This can be visualized as a directed graph, as follows:
+```
+    win7-x64    win7-x86
+       |   \   /    |
+       |   win7     |
+       |     |      |
+    win-x64  |  win-x86
+          \  |  /
+            win
+             |
+            any
+```
+As such, best RID, when evaluating for win7-x64 would be:`win7-x64`, `win7`, `win-x64`, `win`, `any`
+Similarly, when evaluating for `win-x64`: `win-x64`, `win`, `any`
+Note that `win7` comes before `win-x64` due to the import for `win7` appearing before the import for `win-x64` in document order.
+
+###RID-qualified assets are preferred
+NuGet will always prefer a RID-qualified asset over a RID-less asset.  For example if a package contains
+```
+lib/netcoreapp1.0/foo.dll
+runtimes/win/lib/netcoreapp1.0/foo.dll
+```
+When resolving for netstandard1.0/win7-x64 NuGet will choose `runtimes/win/lib/netcoreapp1.0/foo.dll`.
+
+Additionally, NuGet will always prefer a RID-qualified asset over a RID-less asset, even if the framework is less specific for the RID-qualified asset.
+```
+lib/netstandard1.5/foo.dll
+runtimes/win/lib/netstandard1.0/foo.dll
+```
+When resolving for netstandard1.5/win7-x64 NuGet will choose `runtimes/win/lib/netstandard1.0/foo.dll` over `lib/netstandard1.5/foo.dll` even though `netstandard1.5` is more specific than `netstandard1.0`.
+
+### RID-qualified assets are never used for compile
```

Comment:
What happens if you only have rid-qualified assets? For example, what if you had this?

```
runtimes/unix/lib/netstandard1.0/foo.dll
runtimes/win/lib/netstandard1.0/foo.dll
```